kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/predicates/In.java

```diff
@@ -116,8 +116,11 @@ public String toString() {
   }
 
   public String toString(Schema schema) {
-    return Joiner.on(',').join(
-        Iterables.transform(set, new ToString<T>(schema)));
+    // we want the string this returns to be consistent
+    // ensure the set is sorted before calling toString on it.
+
+    Iterable<String> strings = Iterables.transform(set, new ToString<T>(schema));
+    return Joiner.on(',').join(Sets.newTreeSet(strings));
```

Comment:
When making this change we came across a potential bug we wanted to chat about. After the change a test like the one below will fail 100% of the time (Smith,Jones would always be normalized to Jones,Smith). However, in the current code base this test would fail some of the time, since the ordering of the parameters isn't retained at parse time.

    String originalUri = "view:hive:person?lastName=Smith,Jones";
    String getUriResult = Datasets.load(uri).getUri().toString();
    Assert.assertEquals(originalUri, getUriResult);